### PR TITLE
Feature/custom network add

### PR DESCRIPTION
**Pull request by svojsu**
Add/Edit feature for custom and known networks

### CODE REVIEW

File: novawallet/Common/Model/ChainRegistry/LocalChain/ChainModel.swift [hunk 1]
```diff
@@ -533,3 +555,75 @@ extension ChainModel {
         return Set(priceIds)
     }
 }
+
+extension ChainModel {
+    static func partialCustomChainModel(
```
Comment (ERussel) [inline novawallet/Common/Model/ChainRegistry/LocalChain/ChainModel.swift:560]: Do we still need this if we are using separate model for partial chain?

File: novawallet/Modules/NetworkManagement/CustomNetwork/Presenter/CustomNetworkBasePresenter.swift [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+extension CustomNetworkBasePresenter: CustomNetworkBaseInteractorOutputProtocol {
+    func didReceive(_ error: CustomNetworkBaseInteractorError) {
+        let close = R.string.localizable.commonClose(
+            preferredLanguages: selectedLocale.rLanguages
+        )
+        
+        let errorContent = error.toErrorContent(for: selectedLocale)
+        
+        wireframe.present(
```
Comment (ERussel) [inline novawallet/Modules/NetworkManagement/CustomNetwork/Presenter/CustomNetworkBasePresenter.swift:222]: I think wireframe.present(error:...) is a shortcut for the error content presentable errors

**Review by ERussel — APPROVED** [review]

File: novawallet/Common/Model/ChainRegistry/LocalChain/ChainModel.swift (hunk 1 above)
Comment (svojsu) [reply novawallet/Common/Model/ChainRegistry/LocalChain/ChainModel.swift:560]: No, we don't. Forgot to cut it out. Thanks!

File: novawallet/Modules/NetworkManagement/Traits/CustomNetworkSetupTrait.swift [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+        )
+        
+        return wrapper
+    }
+}
+
+private extension CustomNetworkSetupTrait {
+    func createFillPartialChainWrapper(
+        partialChain: PartialCustomChainModel,
+        rawRuntimeFetchFactory: any RuntimeFetchOperationFactoryProtocol,
```
Comment (ERussel) [inline novawallet/Modules/NetworkManagement/Traits/CustomNetworkSetupTrait.swift:72]: Also, here why do we need to use any?

File: novawallet/Modules/NetworkManagement/Traits/CustomNetworkSetupTrait.swift [hunk 4]
```diff
@@ -0,0 +1,279 @@
+import Foundation
+import SubstrateSdk
+import Operation_iOS
+
+protocol CustomNetworkSetupTrait: NetworkNodeCorrespondingTrait {
```
Comment (ERussel) [inline novawallet/Modules/NetworkManagement/Traits/CustomNetworkSetupTrait.swift:5]: I think a separate operation factory suits better here then trait as we create an operation that needs a lot of parameters. I would initialise the factory with parameters and have I single createWrapper() method

File: novawallet/Modules/NetworkManagement/Traits/CustomNetworkSetupTrait.swift [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+        connection: ChainConnection
+    ) -> CompoundOperationWrapper<PartialCustomChainModel> {
+        let propertiesOperation: BaseOperation<SystemProperties>? = chain.isEthereumBased && chain.noSubstrateRuntime
+            ? nil
+            : systemPropertiesOperationFactory.createSystemPropertiesOperation(connection: connection)
+        
+        let assetOperation = ClosureOperation<PartialCustomChainModel> {
+            let properties = try propertiesOperation?.extractNoCancellableResultData()
+            let defaultEVMAssetPrecision: UInt16 = 18
+            let precision = properties?.tokenDecimals.first ?? defaultEVMAssetPrecision
```
Comment (ERussel) [inline novawallet/Modules/NetworkManagement/Traits/CustomNetworkSetupTrait.swift:196]: I would better separate the evm logic from substrate. Just to be sure we don't accedently set improper precision for substrate token

**Review by ERussel — APPROVED** [review]

File: novawallet/Modules/NetworkManagement/Traits/CustomNetworkSetupTrait.swift (hunk 3 above)
Comment (svojsu) [reply novawallet/Modules/NetworkManagement/Traits/CustomNetworkSetupTrait.swift]: got rid of it

File: novawallet/Modules/NetworkManagement/Traits/CustomNetworkSetupTrait.swift (hunk 5 above)
Comment (svojsu) [reply novawallet/Modules/NetworkManagement/Traits/CustomNetworkSetupTrait.swift]: done

File: novawallet/Modules/NetworkManagement/Traits/CustomNetworkSetupTrait.swift (hunk 4 above)
Comment (svojsu) [reply novawallet/Modules/NetworkManagement/Traits/CustomNetworkSetupTrait.swift:5]: Agree, will do it as a separate PR

**Review by ERussel — APPROVED** [review]

**Pull request merged**